pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -93,61 +94,19 @@ func machinesEqual(m1 *clusterv1.Machine, m2 *clusterv1.Machine) bool {
 	return m1.Name == m2.Name && m1.Namespace == m2.Namespace
 }
 
-// isNodeJoin determines if a machine, in scope, should join of the cluster.
-// TODO: Make this thread safe kubernetes-sigs/cluster-api#925
-// https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/745#discussion_r280506890
-func (a *Actuator) isNodeJoin(scope *actuators.MachineScope, controlPlaneMachines []*clusterv1.Machine) (bool, error) {
-	switch set := scope.Machine.ObjectMeta.Labels["set"]; set {
-	case "node":
-		// Worker machines, not part of the controlplane, will always join the cluster.
-		return true, nil
-	case "controlplane":
-		// Controlplane machines will join the cluster if the cluster has an existing control plane.
-		controlplaneExists := false
-		var err error
-
-		var sharedScope *actuators.MachineScope
-
-		for _, cm := range controlPlaneMachines {
-			if sharedScope == nil {
-				sharedScope, err = actuators.NewMachineScope(actuators.MachineScopeParams{
-					Machine:    cm,
-					Cluster:    scope.Cluster,
-					Client:     a.clusterClient,
-					Logger:     a.log,
-					AWSClients: scope.AWSClients,
-				})
-				if err != nil {
-					return false, errors.Wrapf(err, "failed to create machine scope for %s/%s", cm.Namespace, cm.Name)
-				}
-			}
-			sharedScope.Machine = cm
-			ec2svc := ec2.NewService(sharedScope.Scope)
-
-			controlplaneExists, err = ec2svc.MachineExists(sharedScope)
-			if err != nil {
-				return false, errors.Wrapf(err, "failed to verify existence of machine %s/%s", sharedScope.Machine.Namespace, sharedScope.Machine.Name)
-			}
-
-			if controlplaneExists {
-				return controlplaneExists, err
-			}
-		}
-
-		a.log.V(2).Info("Will machine join the controlplane", "machine-name", scope.Machine.Name, "machine-namespace", scope.Machine.Name, "should-join-control-plane", controlplaneExists)
-		return controlplaneExists, err
-
-	default:
-		return false, errors.Errorf("Unknown value %q for label `set` on machine %q", set, scope.Machine.Name)
-	}
-}
-
 // Create creates a machine and is invoked by the machine controller.
 func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	if cluster == nil {
 		return errors.Errorf("missing cluster for machine %s/%s", machine.Namespace, machine.Name)
 	}
-	a.log.Info("Creating machine in cluster", "machine-name", machine.Name, "machine-namespace", machine.Namespace, "cluster-name", cluster.Name)
+
+	log := a.log.WithValues("machine-name", machine.Name, "namespace", machine.Namespace, "cluster-name", cluster.Name)
+	log.Info("Processing machine creation")
+
+	if cluster.Annotations[v1alpha1.AnnotationClusterInfrastructureReady] != "true" {
+		log.Info("Cluster infrastructure is not ready yet - requeuing machine")
+		return &controllerError.RequeueAfterError{RequeueAfter: 5 * time.Second}
```

Comment:
Should we wait a little longer than 5s here? Usually it might take a while until the Cluster is fully ready.

Reply:
- I doubt 5s is harmful, except for the fact this logs at the Info level and you'll see this every 5s.
- We can increase it, but I don't want it to be too long.
- We should probably create a constant.
- We may want multiple constants, if we decide to have different requeue times.

Reply:
Related to https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/392